crates/net/eth-wire/src/ethstream.rs

```diff
@@ -212,101 +223,109 @@ where
         ProtocolBroadcastMessage::from(item).encode(&mut bytes);
         let bytes = bytes.freeze();
 
-        self.inner.start_send_unpin(bytes)?;
+        if self.p2p_tx.try_send(bytes).is_err() {}
 
         Ok(())
     }
 }
 
-impl<S, E> Stream for EthStream<S>
-where
-    S: Stream<Item = Result<BytesMut, E>> + Unpin,
-    EthStreamError: From<E>,
-{
-    type Item = Result<EthMessage, EthStreamError>;
-
-    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
-        let this = self.project();
-        let res = ready!(this.inner.poll_next(cx));
-        let bytes = match res {
-            Some(Ok(bytes)) => bytes,
-            Some(Err(err)) => return Poll::Ready(Some(Err(err.into()))),
-            None => return Poll::Ready(None),
-        };
-
-        if bytes.len() > MAX_MESSAGE_SIZE {
-            return Poll::Ready(Some(Err(EthStreamError::MessageTooBig(bytes.len()))))
-        }
-
-        let msg = match ProtocolMessage::decode_message(*this.version, &mut bytes.as_ref()) {
-            Ok(m) => m,
-            Err(err) => {
-                tracing::debug!("decode error: msg={bytes:x}");
-                return Poll::Ready(Some(Err(err)))
-            }
-        };
-
-        if matches!(msg.message, EthMessage::Status(_)) {
-            return Poll::Ready(Some(Err(EthStreamError::EthHandshakeError(
-                EthHandshakeError::StatusNotInHandshake,
-            ))))
-        }
-
-        Poll::Ready(Some(Ok(msg.message)))
-    }
+#[derive(From)]
+enum Message {
+    Decoded(EthMessage),
+    Encoded(Bytes),
```

Comment:
This is a great change, this would allow us to send broadcasts more efficiently.

however, only newBlock messages would really benefit from this and these are deprecated.

while emitting transactions are broadcast message types, the actual message content depends on the node (we need to ensure that we only send txs to a node that it hasn't seen before), so these messages are still more or less unique...